flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedClientHandler.java

```diff
@@ -152,6 +170,26 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
 		}
 	}
 
+	@Override
+	public void userEventTriggered(ChannelHandlerContext ctx, Object msg) throws Exception {
+		if (msg instanceof RemoteInputChannel) {
+			boolean triggerWrite = inputChannelsWithCredit.isEmpty();
```

Comment:
how about some small comment as in `PartitionRequestQueue`? Something like
```
			// Queue an input channel for available credits announcement.
			// If the queue is empty, we try to trigger the actual write.
			// Otherwise this will be handled by the
			// writeAndFlushNextMessageIfPossible calls.
```